### PR TITLE
Implement Open File in Launcher.  Fixes #601 & #425

### LINKED ISSUE #601: Opening files in the default application
The idea is to provide a method for opening files in the application by default. 
For example, in my application I need to download files of different types, for example pdf, docx, pictures, and I’ll like to be able to open them with applications already installed, for example Acrobat for pdf, but for the system to choose.

### PR DESCRIPTION
**Pull request by jamesmontemagno**
### Description of Change ###

Describe your changes here. 

### Bugs Fixed ###

- Related to issue #601 & #425

Provide links to issues here. Ensure that a GitHub issue was created for your feature or bug fix before sending PR.

### API Changes ###

List all API changes here (or just put None), example:

Added: 
 
- OpenFileRequest
- ReadOnlyFile: FileBase
- Launcher.OpenAsync(OpenFileRequest)

Changed:

Removed "AttachmentName" from the files as it shouldn't be set.

### PR Checklist ###

- [x] Has tests (if omitted, state reason in description)
- [x] Has samples (if omitted, state reason in description)
- [x] Rebased on top of master at time of PR
- [x] Changes adhere to coding standard
- [x] Updated documentation ([see walkthrough](https://github.com/xamarin/Essentials/wiki/Documenting-your-code-with-mdoc))

### CODE REVIEW

**Comment by jamesmontemagno** [comment]
Still need to test UWP and iOS, but should be close

File: Xamarin.Essentials/Launcher/Launcher.shared.cs [hunk 1]
```diff
@@ -36,5 +37,81 @@ public static Task OpenAsync(Uri uri)
...
+
+        public LauncherFile(FileBase file)
+            : base(file)
+        {
+            ExperimentalFeatures.VerifyEnabled(ExperimentalFeatures.OpenFileRequest);
+        }
+
+        string attachmentName;
+
+        public string AttachmentName
```
Comment (mattleibow) [inline Xamarin.Essentials/Launcher/Launcher.shared.cs]: I see this is the third time we have this implemented (Share and Email). We should probably move the implementation to the base (and make it protected if we don't want it visible). We can call it something like `FileNameOverride` and the implementations can just use this value under the hood.

But, I don't see this being used here at all, so I am not sure if this even needs to be here for this case.

Comment (jamesmontemagno) [reply Xamarin.Essentials/Launcher/Launcher.shared.cs]: Fixed!

**Comment by RhomGit** [comment]
Hi @jamesmontemagno , firstly thanks for this addition to Xamarin Essentials!

I am testing the latest nuget (v1.2.0-pre) and I don't think the OpenFileRequest is working on iOS. It works perfectly on Android, and the Share seems to work perfectly in both Android and iOS.


`                string fn = this.FileName_FullIncludingPath;`
`                var uri = new Uri(fn);`
`                var supportsUri = await Launcher.CanOpenAsync(uri.Scheme + @"://");`
`                if (supportsUri)`
`                    await Launcher.OpenAsync(new OpenFileRequest(this.fileCaption, new ReadOnlyFile(fn)));`

The above code executes fine but the OpenAsync call just doesn't seem to do anything on iOS. I have tried .png and .pdf files.

Thanks again,
Denis

**Pull request merged**